FXyz-Core/src/main/java/org/fxyz/shapes/composites/PolyLine3D.java

```diff
@@ -27,72 +27,137 @@
-        this.color = color;
-        setDepthTest(DepthTest.ENABLE);        
-        mesh  = new TriangleMesh();
-        //add each point. For each point add another point shifted on Z axis by width
-        //This extra point allows us to build triangles later
-        for(Point3D point: points) {
-            mesh.getPoints().addAll(point.x,point.y,point.z);
-            mesh.getPoints().addAll(point.x,point.y,point.z+width);
-        }
-        //add dummy Texture Coordinate
-        mesh.getTexCoords().addAll(0,0); 
-        //Now generate trianglestrips for each line segment
-        for(int i=2;i<points.size()*2;i+=2) {  //add each segment
-            //Vertices wound counter-clockwise which is the default front face of any Triange
-            //These triangles live on the frontside of the line facing the camera
-            mesh.getFaces().addAll(i,0,i-2,0,i+1,0); //add primary face
-            mesh.getFaces().addAll(i+1,0,i-2,0,i-1,0); //add secondary Width face
-            //Add the same faces but wind them clockwise so that the color looks correct when camera is rotated
-            //These triangles live on the backside of the line facing away from initial the camera
-            mesh.getFaces().addAll(i+1,0,i-2,0,i,0); //add primary face
-            mesh.getFaces().addAll(i-1,0,i-2,0,i+1,0); //add secondary Width face
-        }
-        //Need to add the mesh to a MeshView before adding to our 3D scene 
-        meshView = new MeshView(mesh);
-        meshView.setDrawMode(DrawMode.FILL);  //Fill so that the line shows width
-        material = new PhongMaterial(color);
-        material.setDiffuseColor(color);
-        material.setSpecularColor(color);
-        meshView.setMaterial(material); 
-        //Make sure you Cull the Back so that no black shows through
-        meshView.setCullFace(CullFace.BACK);
-
-        //Add some ambient light so folks can see it
-        AmbientLight light = new AmbientLight(Color.WHITE);
-        light.getScope().add(meshView);
-        getChildren().add(light);
-        getChildren().add(meshView);           
-    }
+package org.fxyz.shapes.composites;
+
+import java.util.List;
+import javafx.scene.AmbientLight;
+import javafx.scene.DepthTest;
+import javafx.scene.Group;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.PhongMaterial;
+import javafx.scene.shape.CullFace;
+import javafx.scene.shape.DrawMode;
+import javafx.scene.shape.MeshView;
+import javafx.scene.shape.TriangleMesh;
+import org.fxyz.geometry.Point3D;
+
+/**
+ *
+ * @author Sean
+ */
+public class PolyLine3D extends Group {
+    
+    public List<Point3D> points;
+    public int width = 2;
+    public Color color = Color.WHITE;
+    private TriangleMesh mesh;
+    public MeshView meshView;
+    public PhongMaterial material;
+    public static enum LineType {RIBBON, TRIANGLE};
+    
+    //Creates a Ribbon PolyLine3D
+    public PolyLine3D(List<Point3D> points, int width, Color color) {
```

Comment:
Oh crap yes you are right. I forgot to fix that. Will fix and issue new pull request